gui/app/(tabs)/registration.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState } from 'react';
+import { View, Text, TextInput, StyleSheet, TouchableOpacity, Image } from 'react-native';
+import * as ImagePicker from 'expo-image-picker';
+
+interface RegistrationFormData {
+    firstName: string;
+    lastName: string;
+    username: string;
+    email: string;
+    password: string;
+    profilePicture: string | null;
+}
+
+const RegistrationScreen: React.FC = () => {
+    const [formData, setFormData] = useState<RegistrationFormData>({
+        firstName: '',
+        lastName: '',
+        username: '',
+        email: '',
+        password: '',
+        profilePicture: null,
+    });
+
+    const handleChoosePhoto = async () => {
+        const { status } = await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (status !== 'granted') {
+            alert('Sorry, we need camera roll permissions to make this work!');
+            return;
+        }
+
+        const result = await ImagePicker.launchImageLibraryAsync({
+            mediaTypes: ImagePicker.MediaTypeOptions.Images,
+            allowsEditing: true,
+            aspect: [4, 4],
+            quality: 1,
+        });
+
+        if (!result.canceled) {
+            setFormData({ ...formData, profilePicture: result.assets[0].uri });
+        }
+    };
+
+    const handleFirstNameChange = (text: string) => {
+        setFormData({ ...formData, firstName: text });
+    };
+
+    const handleLastNameChange = (text: string) => {
+        setFormData({ ...formData, lastName: text });
+    };
+
+    const handleUserNameChange = (text: string) => {
+        setFormData({ ...formData, username: text });
+    };
+
+    const handleEmailChange = (text: string) => {
+        setFormData({ ...formData, email: text });
+    };
+
+    const handlePasswordChange = (text: string) => {
+        setFormData({ ...formData, password: text });
+    };
+
+    const handleProfilePictureChange = (uri: string | null) => {
+        setFormData({ ...formData, profilePicture: uri });
+    };
+
+    const handleSubmit = () => {
+        console.log(formData);
```

Comment:
This can just be a POST request to /api/user/create, using the `formData` as the body.

Reply:
yes .I agree